tests/0102-static_group_rebalance.c

```diff
@@ -0,0 +1,217 @@
+                 rd_kafka_err2name(err),
+                 parts->cnt);
+
+        switch (err)
+        {
+        case RD_KAFKA_RESP_ERR__ASSIGN_PARTITIONS:
+                rk_c->assign_cnt++;
+
+                rd_kafka_assign(rk, parts);
+
+                TEST_ASSERT(rk_c->assign_cnt <= rk_c->max_rebalance_cnt,
+                            "%s rebalanced %d times, max was %d",
+                            rd_kafka_name(rk),
+                            rk_c->assign_cnt, rk_c->max_rebalance_cnt);
+                break;  
+
+        case RD_KAFKA_RESP_ERR__REVOKE_PARTITIONS:
+                rd_kafka_assign(rk, NULL);
+                break;
+
+        default:
+                TEST_FAIL("rebalance failed: %s", rd_kafka_err2str(err));
+                break;
+        }
+
+}
+
+#define _CONSUMER_CNT 2
+int main_0102_static_group_rebalance (int argc, char **argv) {
+        rd_kafka_conf_t *conf;
+
+        const int msgcnt = 100;
+        const char *topic = test_mk_topic_name("0101_static_group_rebalance", 1);
+        _consumer_t c[_CONSUMER_CNT] = RD_ZERO_INIT;
+
+        test_conf_init(&conf, NULL, 60);
+
+        test_conf_set(conf, "session.timeout.ms", "1000");
+        test_conf_set(conf, "auto.offset.reset", "earliest");
+        test_conf_set(conf, "topic.metadata.refresh.interval.ms", "500");
+
+        rd_kafka_conf_set_opaque(conf, &c[0]);
+        test_conf_set(conf, "group.instance.id", "consumer1");
+        c[0].rk = test_create_consumer("static", &rebalance_cb,
+                                       rd_kafka_conf_dup(conf), NULL);
+
+        rd_kafka_conf_set_opaque(conf, &c[1]);
+        test_conf_set(conf, "group.instance.id", "consumer2");
+        c[1].rk = test_create_consumer("static", &rebalance_cb,
+                                       rd_kafka_conf_dup(conf), NULL);
+
+        c[0].max_rebalance_cnt = 1;
+        c[1].max_rebalance_cnt = 1;
+
+        test_consumer_subscribe(c[0].rk, rd_strdup(tsprintf("^%s.*", topic)));
+        test_consumer_subscribe(c[1].rk, rd_strdup(tsprintf("^%s.*", topic)));
+
+        test_create_topic(NULL, topic, 3, 1);
+        test_produce_msgs_easy(topic, test_id_generate(), 0, msgcnt);
+
+        TEST_SAY("Waiting for partition assignment\n");
+        await_assignment(&c[0]);
+        await_assignment(&c[1]);
+
+        TEST_SAY("Bouncing c[1] instance.\n");
+        test_consumer_close(c[1].rk);
+        rd_kafka_destroy(c[1].rk);
+
+        /** 
+         * Removeing and adding a new member with the same group instance id 
+         * should not prompt a rebalance for the original remaining member.
+         */
+        c[1].max_rebalance_cnt++;
+        c[1].rk = test_create_consumer("static", &rebalance_cb,
+                                       conf, NULL);
+        test_consumer_subscribe(c[1].rk, rd_strdup(tsprintf("^%s.*", topic)));
+
+        /* Wait for c[1] to rejoin, serve c[0] ops */
+        await_assignment(&c[1]);
+        do_consume(&c[0], 5/*5s*/);
```

Comment:
we typically use millisecond timeouts.

Reply:
do_consume uses seconds. I've reduced the timeout